viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
 # ------------- WAVE 1 --------------------
 
+# from tkinter.tix import InputOnly
+
+
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+        # validate input
+    new_dict = {}
+    new_dict["title"] = title
+    new_dict["genre"] = genre
+    new_dict["rating"] = rating
+    # if not new_dict["key"]
+    # for key in new_dict:
+    #     if not new_dict[key]:
+    #         return None
+    return new_dict
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            i = user_data["watchlist"].index(movie)
+            user_data["watchlist"].pop(i)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
+
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0.0
+
+    rating_list=[]
+    for movie in user_data["watched"]:
+        if movie["rating"]> 0:
+            rating_list.append(movie["rating"])
+    
+    rating_sum=sum(rating_list)
+    ave_rating=rating_sum/len(rating_list)
+    
+    return ave_rating
+    
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    frequent_genre_counter={}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in frequent_genre_counter:
+            frequent_genre_counter[genre]+=1
+        else:
+            frequent_genre_counter[genre]=1
+    max_genre = max(frequent_genre_counter.values())
+    for genre in frequent_genre_counter:
+        if frequent_genre_counter[genre]== max_genre:
+            return genre
+    

```

Comment:
Good job with this function! One word about using `max`. When pulling in code from beyond what we've talked about, it's great if you could include a comment about the research that you did, and how you think it works. And it's still worth implementing similar functionality a few times yourself, as it can help practice building our fundamental skills. For now, our goal should be getting as much coding practice as possible!